src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -123,6 +124,8 @@ void GcsJobManager::ClearJobInfos(const JobID &job_id) {
   for (auto &listener : job_finished_listeners_) {
     listener(std::make_shared<JobID>(job_id));
   }
+  // Clear cache.
+  RAY_UNUSED(cached_job_configs_.erase(job_id));
```

Comment:
Hum, this clearing makes `test_placement_group_2.py::test_automatic_cleanup_detached_actors` failed. The reason is PG will use `get_ray_namespace(job_id1)` after job_id1 finished.
@wuisawesome @clay4444 CC 
The stack is:
```shell
[2021-12-28 17:56:18,501 I 63827 7604794] gcs_actor_manager.cc:990: Actor created successfully, actor id = 60ff9e9b3a4224e2b337e2b602000000, job id = 02000000
[2021-12-28 17:56:18,595 C 63827 7604794] gcs_job_manager.cc:176:  Check failed: it != cached_job_configs_.end() Couldn't find job with id: 02000000
*** StackTrace Information ***
    ray::GetCallTrace()
    ray::SpdLogMessage::Flush()
    ray::SpdLogMessage::~SpdLogMessage()
    ray::RayLog::~RayLog()
    ray::gcs::GcsJobManager::GetRayNamespace()
    std::__1::__function::__func<>::operator()()
    ray::gcs::GcsPlacementGroupManager::HandleCreatePlacementGroup()
    ray::rpc::ServerCallImpl<>::HandleRequestImpl()
    boost::asio::detail::completion_handler<>::do_complete()
    boost::asio::detail::scheduler::do_run_one()
    boost::asio::detail::scheduler::run()
    boost::asio::io_context::run()
    main
    start
    0x0
```

Reply:
If we don't clear it, it will waste memory of GCS, which maybe an issue in large scale cluster.